halo2_proofs/src/arithmetic.rs

```diff
@@ -408,6 +408,32 @@ use rand_core::OsRng;
 #[cfg(test)]
 use crate::pasta::Fp;
 
+#[test]
+fn test_eval_polynomial() {
+    for k in 3..10 {
+        let mut eval = Fp::zero();
+        let mut exp = Fp::one();
+        let point = Fp::random(OsRng);
+        let poly = (0..(1 << k)).map(|_| Fp::random(OsRng)).collect::<Vec<_>>();
```

Comment:
We can use a proptest for this, with `arb_fp` and `arb_poly` prop_compose strategies.